File: modulo1/programasGabarito/ex1zeroMaquina.f90
program zero_maquina
  implicit none
  real(8) :: eps

  eps = 1.0d0
  do while (1.0d0 + eps /= 1.0d0)
     eps = eps / 2.0d0
  end do

  eps = eps * 2.0d0
  print *, "Zero de máquina aproximado: ", eps
end program zero_maquina

File: modulo1/programasGabarito/ex1zeroMaquina.cpp
#include <iostream>
using namespace std;

int main() {
    double eps = 1.0;
    while (1.0 + eps != 1.0) {
        eps /= 2.0;
    }
    eps *= 2.0;
    cout.precision(20);
    cout << "Zero de máquina aproximado: " << eps << endl;
    return 0;
}
